src/compiler.js

```diff
@@ -69,18 +102,168 @@ export class NexeCompiler {
   _configureAsync () {
     return this._runBuildCommandAsync(
       this.env.PYTHON || 'python',
-      [configure, ...this.configure]
+      [configure, ...this.options.configure]
     )
   }
 
   async _buildAsync () {
+    if (this.options.clean) {
+      await this._runBuildCommandAsync(make, ['clean'])
+    }
     await this._configureAsync()
-    return this._runBuildCommandAsync(make, this.make)
+    const buildOptions = isWindows ? this.options.vcBuild : this.options.make
+    await this._runBuildCommandAsync(make, buildOptions)
+    return createReadStream(this._getArtifactLocation())
+  }
+
+  _fetchPrebuiltBinary () {
+    /**
+     * TODO implement hosted builds
+     *  - CircleCI artifacts
```

Comment:
I guess I'm wondering why circle can't do the github release as well for the linux/mac versions, and appveyor do release for windows..?  Why use artifacts at all.  Both side could check for the github release and create/update it accordingly